pkg/detector/detector.go

```diff
@@ -694,6 +695,20 @@ func (d *ResourceDetector) GetReplicaDeclaration(object *unstructured.Unstructur
 			return nil, 0, err
 		}
 		return replicaRequirements, int32(replicas), nil
+	case util.JobKind:
+		replicas, ok, err := unstructured.NestedInt64(object.Object, util.SpecField, util.ParallelismField)
```

Comment:
The replicas might be zero. refer to https://github.com/karmada-io/karmada/issues/893#issuecomment-962429514.

Reply:
Hi @RainbowMango, the replicas come from `.spec.parallelism` field. It cannot be zero. I think you mean `.spec.completions`? That is not `replicas`.

Reply:
See [here](https://github.com/kubernetes/kubernetes/blob/dccaa29e1531b59f4b4b48d0e0c4777cb530c555/pkg/apis/batch/v1/defaults.go#L34-L41)

Reply:
right.